Dockerfile.assisted-service.ocp

```diff
@@ -1,8 +1,4 @@
-# URGENT! ART metadata configuration has a different number of FROMs
-# than this Dockerfile. ART will be unable to build your component or
-# reconcile this Dockerfile until that disparity is addressed.
-# Build binaries
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.14 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.20-openshift-4.14 AS builder
```

Comment:
is this PR blocked by https://github.com/openshift/release/pull/41331 or we just don't want prow to garbage-collect the ``ocp_builder_rhel-9-golang-1.20-openshift-4.14`` images?

Reply:
Yes I've added a comment for that https://github.com/openshift/assisted-service/pull/5361#issuecomment-1637847768

Reply:
Yeah, saw that comment. Just wasn't sure for the exact reason (which of the two reasons mentioned above applies here?)

Reply:
The first one, waiting to have the new image available in the CI config so that we could test the switch here

Reply:
(and btw the bot didn't update as expected the Dockerfile)

Reply:
@osherdp I've double checked and given the current configuration should be ok to land this PR first, in any case it's blocked by  https://issues.redhat.com/browse/ART-7241

Reply:
yes that makes sense as the changes in release repo are mainly image substitutions for performance reasons iiuc
either way I'll start following that ticket to be notified for the follow-ups